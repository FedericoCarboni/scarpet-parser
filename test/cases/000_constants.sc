// 
euler;pi;null;true;false