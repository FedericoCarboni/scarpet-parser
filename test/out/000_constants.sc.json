{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "start": {
      "pos": 4,
      "row": 0,
      "col": 0
    },
    "end": {
      "pos": 28,
      "row": 0,
      "col": 0
    },
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "start": {
        "pos": 4,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 22,
        "row": 0,
        "col": 0
      },
      "operator": ";",
      "lvalue": {
        "kind": "BinaryExpression",
        "start": {
          "pos": 4,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 17,
          "row": 0,
          "col": 0
        },
        "operator": ";",
        "lvalue": {
          "kind": "BinaryExpression",
          "start": {
            "pos": 4,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 12,
            "row": 0,
            "col": 0
          },
          "operator": ";",
          "lvalue": {
            "kind": "Constant",
            "start": {
              "pos": 4,
              "row": 0,
              "col": 0
            },
            "end": {
              "pos": 9,
              "row": 0,
              "col": 0
            },
            "name": "euler"
          },
          "rvalue": {
            "kind": "Constant",
            "start": {
              "pos": 10,
              "row": 0,
              "col": 0
            },
            "end": {
              "pos": 12,
              "row": 0,
              "col": 0
            },
            "name": "pi"
          }
        },
        "rvalue": {
          "kind": "Constant",
          "start": {
            "pos": 13,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 17,
            "row": 0,
            "col": 0
          },
          "name": "null"
        }
      },
      "rvalue": {
        "kind": "Constant",
        "start": {
          "pos": 18,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 22,
          "row": 0,
          "col": 0
        },
        "name": "true"
      }
    },
    "rvalue": {
      "kind": "Constant",
      "start": {
        "pos": 23,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 28,
        "row": 0,
        "col": 0
      },
      "name": "false"
    }
  }
}