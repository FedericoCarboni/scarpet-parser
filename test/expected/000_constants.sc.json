{
    "errors": [],
    "warnings": [],
    "root": {
        "kind": "BinaryExpression",
        "start": {
            "pos": 4,
            "row": 1,
            "col": 0
        },
        "end": {
            "pos": 28,
            "row": 1,
            "col": 24
        },
        "operator": ";",
        "lvalue": {
            "kind": "BinaryExpression",
            "start": {
                "pos": 4,
                "row": 1,
                "col": 0
            },
            "end": {
                "pos": 22,
                "row": 1,
                "col": 18
            },
            "operator": ";",
            "lvalue": {
                "kind": "BinaryExpression",
                "start": {
                    "pos": 4,
                    "row": 1,
                    "col": 0
                },
                "end": {
                    "pos": 17,
                    "row": 1,
                    "col": 13
                },
                "operator": ";",
                "lvalue": {
                    "kind": "BinaryExpression",
                    "start": {
                        "pos": 4,
                        "row": 1,
                        "col": 0
                    },
                    "end": {
                        "pos": 12,
                        "row": 1,
                        "col": 8
                    },
                    "operator": ";",
                    "lvalue": {
                        "kind": "Constant",
                        "start": {
                            "pos": 4,
                            "row": 1,
                            "col": 0
                        },
                        "end": {
                            "pos": 9,
                            "row": 1,
                            "col": 5
                        },
                        "name": "euler"
                    },
                    "rvalue": {
                        "kind": "Constant",
                        "start": {
                            "pos": 10,
                            "row": 1,
                            "col": 6
                        },
                        "end": {
                            "pos": 12,
                            "row": 1,
                            "col": 8
                        },
                        "name": "pi"
                    },
                    "comment": "// \n"
                },
                "rvalue": {
                    "kind": "Constant",
                    "start": {
                        "pos": 13,
                        "row": 1,
                        "col": 9
                    },
                    "end": {
                        "pos": 17,
                        "row": 1,
                        "col": 13
                    },
                    "name": "null"
                },
                "comment": "// \n"
            },
            "rvalue": {
                "kind": "Constant",
                "start": {
                    "pos": 18,
                    "row": 1,
                    "col": 14
                },
                "end": {
                    "pos": 22,
                    "row": 1,
                    "col": 18
                },
                "name": "true"
            },
            "comment": "// \n"
        },
        "rvalue": {
            "kind": "Constant",
            "start": {
                "pos": 23,
                "row": 1,
                "col": 19
            },
            "end": {
                "pos": 28,
                "row": 1,
                "col": 24
            },
            "name": "false"
        },
        "comment": "// \n"
    }
}